core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala

```diff
@@ -279,16 +279,28 @@ trait H2OContextRestAPIUtils extends H2OContextUtils {
         case unknown => throw new IllegalArgumentException(s"Unsupported HTTP request type $unknown")
       }
       getCredentials(conf).foreach(creds => request.setHeader(HttpHeaders.AUTHORIZATION, creds))
-      httpClient.execute(request).getEntity.getContent
+      val result = httpClient.execute(request)
+      val statusCode = result.getStatusLine.getStatusCode
+      statusCode match {
+        case 401 => throw new RestApiException("Unauthorized")
+        case _ =>
+      }
+      result.getEntity.getContent
     } catch {
+      case _: RestApiException => throw new H2OClusterNodeNotReachableException(
```

Comment:
WDYT about passing the original `RestApiException` to `H2OClusterNodeNotReachableException` as the second argument?